typescript/infra/src/utils/sealevel.ts

```diff
@@ -0,0 +1,246 @@
+import {
+  ComputeBudgetProgram,
+  Connection,
+  Keypair,
+  PublicKey,
+  Transaction,
+  TransactionInstruction,
+} from '@solana/web3.js';
+import { resolve } from 'path';
+
+import { SEALEVEL_PRIORITY_FEES } from '@hyperlane-xyz/sdk';
+import { rootLogger } from '@hyperlane-xyz/utils';
+
+import { DeployEnvironment } from '../config/environment.js';
+
+import { getMonorepoRoot, readJSONAtPath } from './utils.js';
+
+export const svmGasOracleConfigPath = (environment: DeployEnvironment) =>
+  resolve(
+    getMonorepoRoot(),
+    `rust/sealevel/environments/${environment}/gas-oracle-configs.json`,
+  );
+
+/**
+ * Build and send a transaction with priority fees and custom confirmation polling
+ * This avoids WebSocket subscription issues with some RPC providers
+ */
+export async function buildAndSendTransaction(
+  connection: Connection,
+  instructions: TransactionInstruction[],
+  signer: Keypair,
+  chain: string,
+): Promise<string> {
+  const tx = new Transaction();
+
+  // Add priority fee if configured
+  const priorityFee = SEALEVEL_PRIORITY_FEES[chain];
+  if (priorityFee) {
+    tx.add(
+      ComputeBudgetProgram.setComputeUnitPrice({
+        microLamports: priorityFee,
+      }),
+    );
+  }
+
+  // Add all instructions
+  for (const instruction of instructions) {
+    tx.add(instruction);
+  }
+
+  // Get recent blockhash
+  const { blockhash } = await connection.getLatestBlockhash('confirmed');
+  tx.recentBlockhash = blockhash;
+  tx.feePayer = signer.publicKey;
+
+  // Sign and send transaction
+  tx.sign(signer);
+  const signature = await connection.sendRawTransaction(tx.serialize(), {
+    skipPreflight: false,
+    maxRetries: 3,
+  });
+
+  // Poll for confirmation using getSignatureStatus instead of confirmTransaction
+  let confirmed = false;
+  let attempts = 0;
+  const maxAttempts = 30; // 30 seconds timeout
+
+  while (!confirmed && attempts < maxAttempts) {
+    await new Promise((resolve) => setTimeout(resolve, 1000)); // Wait 1 second
+    attempts++;
+
+    try {
+      const status = await connection.getSignatureStatus(signature, {
+        searchTransactionHistory: true,
+      });
+
+      if (status.value) {
+        if (status.value.err) {
+          throw new Error(
```

Comment:
We probably want to rethrow this error in the catch block or terminate the polling loop in some other way. If this error state is not supposed to terminate the loop, why do we throw?